R/get_genstate_vax.R

```diff
@@ -0,0 +1,46 @@
+#' Get general population state COVID vaccine data from CDC and Census 2020 18+ pop estimates
+#'
+#' @description Pulls the latest data from the CDC on COVID vaccinations initiated
+#' for adults in a state and then merges alongside a population number from the 18+ 2020
+#' Census population estimates for that state to use for rate calculation.
+#'
+#' @return a data frame with the following columns: State, Date,
+#' General.Initiated, General.Adult.Population
+#'
+#' @examples
+#' \dontrun{
+#'
+#'# get state-level vax data
+#' st_gen_df <- get_genstate_vax()
+#'
+#'
+#' }
+#' @importFrom jsonlite read_json
+#' @export
+
+get_genstate_vax <- function(){
+
+    adultpop_df <- "https://www2.census.gov/programs-surveys/popest/datasets/" %>%
+        stringr::str_c("2010-2020/state/asrh/sc-est2020-18+pop-res.csv") %>%
+        readr::read_csv(col_types = readr::cols()) %>%
+        select(State = NAME,
+               General.Adult.Population = POPEST18PLUS2020) %>%
+        mutate(State = ifelse(State == "United States", "National", State))
+
+    raw_cdc_vax <- str_c(
+        "https://covid.cdc.gov/covid-data-tracker/COVIDData/",
+        "getAjaxData?id=vaccination_data") %>%
+        jsonlite::read_json(simplifyVector = TRUE)
+
+    gen_vax_df <- as_tibble(raw_cdc_vax$vaccination_data) %>%
+        mutate(General.Initiated = Administered_Dose1_Recip_18Plus)  %>%
```

Comment:
Thoughts on also adding `General.Completed` here? I think it corresponds to `Series_Complete_18Plus` in the CDC data (cross-checked a few numbers and this seemed right, but always good to verify)!

Reply:
Good idea, thanks! Adding this in now